### DISCUSSION TITLE
The behavior of the p:confirmDialog is inconsistent with expectations.

### DISCUSSION THREAD

**Discussion by WhaleIsland** [primefaces/community#3132] [PrimeFaces] [answered]
I used similar code to implement confirmation and execute certain processing, with two possible results: returning to the current view or navigating to another view.
```
<p:commandButton value="Confirm" action="#{confirmView.confirm}" update="message" styleClass="mr-2" icon="pi pi-check">
    <p:confirm header="Confirmation" message="Are you sure you want to proceed?" icon="pi pi-exclamation-triangle"/>
</p:commandButton>
```
Added a message using `FacesContext.getCurrentInstance().addMessage()`.

- When null is returned in confirmView.confirm, the growl is displayed.
- When other view IDs are returned in confirmView.confirm, the growl does not display, and the elements on the screen are not re-rendered.

There is the following information in the console:
```
[Violation] 'load' handler took 585ms
[Violation] Forced reflow while executing JavaScript took 39ms
```
But from a visual perspective, it seems there is no reflow.

I tried using `FacesContext.getCurrentInstance().getExternalContext().getFlash().setKeepMessages(true)`, the result is still the same, but after performing other actions on the result page, the growl is displayed.

Dependency versions:
- jakarta.jakartaee-api:jar:9.1.0
- primefaces:jar:jakarta:13.0.0

Should I try some other approaches?

**Comment by melloware** [comment]
can you provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project?  It is the only way developers can debug your problem to help.

I need to see what you are trying to do.

Also as a note FlashMessages use a cookie which does NOT work in LocalHost in most browsers but will work in your real website.  Have you tried it not running `localhost`  . Take a look at this PhaseListener which gets around the Flash localhost problem: https://github.com/melloware/quarkus-faces/blob/main/src/main/java/org/primefaces/showcase/util/MultiPageMessagesSupport.java

**Reply by WhaleIsland** [reply, marked as answer]
I tried writing an example in PrimeFaces Test, and the FlashMessages display correctly.
There are still some filters and listeners in the development project, so I cannot write a complete executable test case in a short time.

In PrimeFaces Test, the `<p:commandButton>` works as expected. In development project, it doesn't work. This might be due to the use of `<ui:include/>`.
I want to find some methods to update components outside the JSF lifecycle.

**Reply by melloware** [reply]
Ahhh yep if you can boil it down to a PF Test reproducer i can look but other than that I have to assume its a bug in what you are doing.
